packages/darklang/stdlib/list.dark

```diff
@@ -47,9 +47,7 @@ module Darklang =
         match list with
         | [] -> []
         | head :: tail ->
-          Builtin.List.append_v0
-            (PACKAGE.Darklang.Stdlib.List.reverse tail)
-            (Builtin.List.singleton_v0 head)
+          Builtin.List.append_v0 (PACKAGE.Darklang.Stdlib.List.reverse tail) [ head ]
```

Comment:
I think this function is O(N^2)

Reply:
I've tried another implementation but it isn't any better it is O(N^2)
```
let reverse (list: List<'a>) : List<'a> =
   PACKAGE.Darklang.Stdlib.List.fold list [] (fun acc item -> (PACKAGE.Darklang.Stdlib.List.append [item] acc))

```
Should we keep it as a Builtin?

Reply:
I think this could be O(N) if we used push instead of append?

Reply:
I asked chatgpt:

```
let reverseList lst =
    let rec reverseHelper acc = function
        | [] -> acc
        | head :: tail -> reverseHelper (head :: acc) tail
    reverseHelper [] lst
```

Reply:
I think we can't do this because of the recursive lambda issue? I am not sure I'll try